src/modules/commander/Commander.cpp

```diff
@@ -1737,6 +1737,11 @@ Commander::run()
 						tune_mission_ok(true);
 					}
 				}
+
+				if (is_auto_state(internal_state.main_state) && mission_result.finished && land_detector.landed && !armed.armed
```

Comment:
Can you make sure this is only called on disarm transition by adding a check for `was_armed`?
You also don't necessarily have to check the landed flag.

Reply:
It seems that `was_armed` is usually updated sooner than the `mission_result` message is available, so it happens in a previous iteration of the main loop. Would it be OK to check if the armed-disarmed transition happened, for example, in the past 1 second?

Reply:
No, can you move the `was_armed` update further down then?